# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmBuiltins.h
# Commit: 9111720170a2
# Full Hash: 9111720170a283e7c3801856d5c113a2f17caf20
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2022-05-06 09:50:52
# Regressor Bug: 1642412
# File Overlap Count: 7
# Description:
#   Bug 1762171 - wasm: Fix cranelift build for table optimizations. r=yury
#   
#   This commit adds back the builtins that were removed in
#   bug 1642412.
#   
# ==============================================================================

diff -r 2340f31f2559 -r 9111720170a2 js/src/wasm/WasmBuiltins.h
--- a/js/src/wasm/WasmBuiltins.h	Thu May 05 21:49:46 2022 +0000
+++ b/js/src/wasm/WasmBuiltins.h	Thu May 05 21:49:46 2022 +0000
@@ -112,10 +112,11 @@
   TableCopy,
   ElemDrop,
   TableFill,
-  TableGetFunc,
+  TableGet,
   TableGrow,
   TableInit,
-  TableSetFunc,
+  TableSet,
+  TableSize,
   RefFunc,
   RefTest,
   RttSub,
@@ -234,10 +235,11 @@
 extern const SymbolicAddressSignature SASigTableCopy;
 extern const SymbolicAddressSignature SASigElemDrop;
 extern const SymbolicAddressSignature SASigTableFill;
-extern const SymbolicAddressSignature SASigTableGetFunc;
+extern const SymbolicAddressSignature SASigTableGet;
 extern const SymbolicAddressSignature SASigTableGrow;
 extern const SymbolicAddressSignature SASigTableInit;
-extern const SymbolicAddressSignature SASigTableSetFunc;
+extern const SymbolicAddressSignature SASigTableSet;
+extern const SymbolicAddressSignature SASigTableSize;
 extern const SymbolicAddressSignature SASigRefFunc;
 extern const SymbolicAddressSignature SASigPreBarrierFiltering;
 extern const SymbolicAddressSignature SASigPostBarrier;